Mathlib/LinearAlgebra/Projection.lean

```diff
@@ -367,12 +368,11 @@ structure IsProj {F : Type*} [FunLike F M M] (f : F) : Prop where
   map_mem : ∀ x, f x ∈ m
   map_id : ∀ x ∈ m, f x = x
 
-theorem isProj_iff_idempotent (f : M →ₗ[S] M) : (∃ p : Submodule S M, IsProj p f) ↔ f ∘ₗ f = f := by
+theorem isProj_iff_idempotent (f : M →ₗ[S] M) :
+    (∃ p : Submodule S M, IsProj p f) ↔ IsIdempotentElem f := by
```

Comment:
Probably now a better name is `isProj_iff_isIdempotentElem`

Reply:
I've renamed it and added a deprecated alias.